### PR TITLE
fix: Add middleware earlier #1775

### PR DESCRIPTION
**Pull request by abador**
## Related issue(s)
#1775

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

## Further Comments
No docs and tests, itonly changes the order of middlewares

### CODE REVIEW

File: cmd/daemon/serve.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ func ServePublic(r driver.Registry, wg *sync.WaitGroup, cmd *cobra.Command, args
 	c := r.Config(cmd.Context())
 	l := r.Logger()
 	n := negroni.New()
+	n.Use(reqlog.NewMiddlewareFromLogger(l, "public#"+c.SelfPublicURL(nil).String()))
+	n.Use(sqa(ctx, cmd, r))
+	n.Use(r.PrometheusManager())
```
Comment (aeneasr) [inline cmd/daemon/serve.go:84]: Please move this after the for loop below

File: cmd/daemon/serve.go [hunk 2]
```diff
@@ -138,16 +138,16 @@ func ServeAdmin(r driver.Registry, wg *sync.WaitGroup, cmd *cobra.Command, args
 	c := r.Config(ctx)
 	l := r.Logger()
 	n := negroni.New()
+	n.Use(reqlog.NewMiddlewareFromLogger(l, "admin#"+c.SelfPublicURL(nil).String()))
+	n.Use(sqa(ctx, cmd, r))
+	n.Use(r.PrometheusManager())
```
Comment (aeneasr) [inline cmd/daemon/serve.go:143]: Please move this after the for loop below

**Review by aeneasr — CHANGES_REQUESTED** [review]

**Review by aeneasr — CHANGES_REQUESTED** [review]
Actually, I don't think that this is really solving the issue you described. The only thing this does is that the negroni middleware is executed before cleanpath and the csrf middleware!

**Comment by abador** [comment]
> Actually, I don't think that this is really solving the issue you described. The only thing this does is that the negroni middleware is executed before cleanpath and the csrf middleware!

Yeah this also looked a bit weird to me. This shouldn't make much difference, but it does, since the CSRFHandler is created based on the router

**Comment by aeneasr** [comment]
Oh I see, that makes sense! :)

File: cmd/daemon/serve.go [hunk 3]
```diff
@@ -79,9 +79,12 @@ func ServePublic(r driver.Registry, wg *sync.WaitGroup, cmd *cobra.Command, args
 	c := r.Config(cmd.Context())
 	l := r.Logger()
 	n := negroni.New()
+	n.Use(reqlog.NewMiddlewareFromLogger(l, "public#"+c.SelfPublicURL(nil).String()))
+	n.Use(sqa(ctx, cmd, r))
 	for _, mw := range modifiers.mwf {
 		n.UseFunc(mw)
 	}
+	n.Use(r.PrometheusManager())
```
Comment (aeneasr) [inline cmd/daemon/serve.go:87]: I meant to say to move all n.Use after the modifiers :) The modifiers are used only for testing, and it is important for them to execute before all other middleware. Thanks!

Comment (abador) [reply cmd/daemon/serve.go:87]: My bad :) Done

**Review by aeneasr — APPROVED** [review]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

**Pull request merged**